[Bug] TestNewFileStorage/empty_path_uses_default test expects error but code uses default path

## Description

The test `TestNewFileStorage/empty_path_uses_default` in `internal/acme/storage_test.go` expects that `NewFileStorage("", logger)` returns an error and nil storage. However, the implementation uses a default path (`/var/lib/novaedge/certs`) when an empty string is passed, so it returns no error and a valid storage object.

## Expected Behavior
Either:
1. The test should be updated to match the implementation (empty path → default path is valid)
2. Or the implementation should be changed to reject empty paths

## Steps to Reproduce
```bash
go test -v -run TestNewFileStorage ./internal/acme/
```

## Error Output
```
--- FAIL: TestNewFileStorage/empty_path_uses_default (0.00s)
    storage_test.go:55: Error: An error is expected but got nil.
    storage_test.go:56: Error: Expected nil, but got: &acme.FileStorage{basePath:"/var/lib/novaedge/certs", ...}
```

## Context
Discovered while running full test suite inside a Linux container. This is a pre-existing issue unrelated to BPF code generation work.

--- a/internal/acme/storage_test.go
+++ b/internal/acme/storage_test.go
@@ -29,37 +29,32 @@ import (
 )
 
 func TestNewFileStorage(t *testing.T) {
-	tests := []struct {
-		name     string
-		basePath string
-		wantErr  bool
-	}{
-		{
-			name:     "valid path",
-			basePath: filepath.Join(t.TempDir(), "acme-storage"),
-			wantErr:  false,
-		},
-		{
-			name:     "empty path uses default",
-			basePath: "",
-			wantErr:  true, // Will fail because default path requires root permissions
-		},
-	}
+	t.Run("valid path", func(t *testing.T) {
+		logger := zap.NewNop()
+		storage, err := NewFileStorage(filepath.Join(t.TempDir(), "acme-storage"), logger)
+		assert.NoError(t, err)
+		assert.NotNil(t, storage)
+	})
 
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			logger := zap.NewNop()
-			storage, err := NewFileStorage(tt.basePath, logger)
+	t.Run("empty path uses default", func(t *testing.T) {
+		logger := zap.NewNop()
+		storage, err := NewFileStorage("", logger)
+		// Whether this succeeds depends on whether the process can create
+		// the default path (/var/lib/novaedge/certs). As root it succeeds;
+		// as a normal user it fails with a permission error.
+		if err != nil {
+			assert.Nil(t, storage)
+		} else {
+			assert.NotNil(t, storage)
+		}
+	})
 
-			if tt.wantErr {
-				assert.Error(t, err)
-				assert.Nil(t, storage)
-			} else {
-				assert.NoError(t, err)
-				assert.NotNil(t, storage)
-			}
-		})
-	}
+	t.Run("unwritable path returns error", func(t *testing.T) {
+		logger := zap.NewNop()
+		storage, err := NewFileStorage("/proc/nonexistent/acme", logger)
+		assert.Error(t, err)
+		assert.Nil(t, storage)
+	})
 }
 
 func TestNewFileStorageWithNilLogger(t *testing.T) {

